x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/rules_settings_modal.tsx

```diff
@@ -60,6 +60,26 @@ export const RulesSettingsErrorPrompt = memo(() => {
   );
 });
 
+const useResettableState: <T>(
+  initialValue?: T
+) => [T | undefined, boolean, (next: T, shouldUpdateInitialValue?: boolean) => void, () => void] = (
```

Comment:
Do we need this heavy type here?

Reply:
This is basically the equivalent type signature to `useState` with the extra properties added, it allows us to pass the type signature for `RulesSettingsFlappingProperties` and `RulesSettingsQueryDelayProperties` into the hook so that the resulting values are properly typed.

Maybe a bit overengineered for this single use case but I wanted to replicate `useState` as much as possible. If we ever have to pull this hook out into its own module then the type work is already done.